hybrasyl/Monolith.cs

```diff
@@ -138,6 +143,155 @@ private static void SpawnMonster(Monster monster, Map map)
+            {
+                var lootSets = Game.World.WorldData.Values<LootSet>();
+                foreach (var set in _spawn.Loot.Set)
+                {
+                    var lootImportName = set.Name;
+                    var lootImportSet = lootSets.Where(lootSet => lootSet.Name == lootImportName);
+
+                    foreach (var importSet in lootImportSet)
+                    {
+                        foreach (var importTable in importSet.Table)
+                        {
+                            for (int i = 0; i < importTable.Rolls; i++)
+                            {
+                                if (importTable.Chance >= _rng.NextDouble())
+                                {
+                                    spawnLootTable.Add(ConvertLootTable(importTable));
+                                    continue;
+                                }
+                            }
+                        }
+                    }
+                }
+
+            }
+            return spawnLootTable;
+        }
+
+        /// <summary>
+        /// Used to Convert Hybrasyl.LootTable to Creatures.LootTable
+        /// </summary>
+        /// <param name="lootTableToConvert">Hybrasyl.LootTable to Convert</param>
+        /// <returns>Creatures.LootTable</returns>
+        private Creatures.LootTable ConvertLootTable(Hybrasyl.LootTable lootTableToConvert)
+        {
+            var newLootTable = new Creatures.LootTable();
+            newLootTable.Items = new Creatures.LootTableItemList();
+            newLootTable.Items.Items = new List<Creatures.LootItem>();
+
+            newLootTable.Chance = lootTableToConvert.Chance;
+            newLootTable.Gold.Min = lootTableToConvert.Gold.Min;
+            newLootTable.Gold.Max = lootTableToConvert.Gold.Max;
+            newLootTable.Items.Chance = lootTableToConvert.Items.Chance;
+
+            foreach(var items in lootTableToConvert.Items.Items)
+            {
+                var newLootItem = new Creatures.LootItem();
+                newLootItem.Always = items.Always;
+                newLootItem.Max = items.Max;
+                newLootItem.Min = items.Min;
+                newLootItem.Unique = items.Unique;
+                newLootItem.Value = items.Value;
+                newLootItem.Variants = items.Variants;
+
+                newLootTable.Items.Items.Add(newLootItem);
+            }
+            
+            newLootTable.Items.Rolls = lootTableToConvert.Items.Rolls;
+            newLootTable.Rolls = lootTableToConvert.Rolls;
+
+            return newLootTable;
+        }
+
+        /// <summary>
+        /// Calculate the total amount of lootable gold based on the base amount plus any additional from the loot table(s).
+        /// </summary>
+        /// <returns>Gold uint</returns>
+        public uint LootableGold()
+        {
+            uint lootableGold = 0;
+            var spawnLootGold = _spawn.Loot.Gold;
+
+            if (spawnLootGold != null)
```

Comment:
`SpawnLootGold` can never be null (it is always initialized - check `Loot.Designer.cs` line 87), so a lot of this stuff can be expressed more concisely. I pushed an update to your branch to enforce some defaults and update XML structure a bit.

Reply:
Will update this.